### PR TITLE
Added a script triggers overlay

### PR DESCRIPTION
**Pull request by Mailaender**
accessible by debug command.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/World/ActorMap.cs [hunk 1]
```diff
@@ -510,6 +510,11 @@ public int AddCellTrigger(CPos[] cells, Action<Actor> onEntry, Action<Actor> onE
 			return id;
 		}
 
+		public List<CPos> TriggerPositions()
+		{
+			return cellTriggerInfluence.Keys.ToList();
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/World/ActorMap.cs]: Consider returning `IEnumerable<CPos>` and avoiding the `ToList`? Seems like that would be a nicer signature to expose.

File: OpenRA.Mods.Common/Traits/World/TriggerOverlay.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+			if (name == CommandName)
+				Enabled ^= true;
+		}
+
+		IEnumerable<IRenderable> IRenderAnnotations.RenderAnnotations(Actor self, WorldRenderer wr)
+		{
+			if (!Enabled)
+				yield break;
+
+			var triggerPositions = wr.World.ActorMap.TriggerPositions();
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/World/TriggerOverlay.cs]: Suggest `ToHashSet` so if there's lots of triggers the `Contains` checks are not too slow.

File: OpenRA.Mods.Common/Traits/World/TriggerOverlay.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+			foreach (var uv in wr.Viewport.VisibleCellsInsideBounds.CandidateMapCoords)
+			{
+				if (self.World.ShroudObscures(uv))
+					continue;
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/World/TriggerOverlay.cs]: Thought: Do we care about shroud for this - would it be better to show them regardless?

**Review by RoosterDragon — COMMENTED** [review]
Works fine and appears sensible, some minor thoughts added.

File: OpenRA.Mods.Common/Traits/World/TriggerOverlay.cs (hunk 3 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/World/TriggerOverlay.cs]: It looks a bit stupid and all the other debug overlays don't see through shroud.

**Comment by Mailaender** [comment]
Thanks for the review. Updated accordingly.

**Review by RoosterDragon — DISMISSED** [review]

File: OpenRA.Mods.Common/Traits/World/TriggerOverlay.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+		public override object Create(ActorInitializer init) { return new CellTriggerOverlay(this); }
+	}
+
+	public class CellTriggerOverlay : IRenderAnnotations, IWorldLoaded, IChatCommand
+	{
+		const string CommandName = "triggers";
+		const string CommandDesc = "toggles the script triggers overlay.";
+
+		public bool Enabled;
```
Comment (teinarss) [inline OpenRA.Mods.Common/Traits/World/TriggerOverlay.cs]: Can be private

Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/World/TriggerOverlay.cs]: Made private.

**Comment by teinarss** [comment]
Also needs a rebase

**Comment by Mailaender** [comment]
Rebased after https://github.com/OpenRA/OpenRA/pull/19895

**Review by teinarss — APPROVED** [review]

**Comment by teinarss** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/c9fe5938111dcec481d6689f007b34d4e973373e...294e8b103714b5244c2e1a1059e4178e2bddbdba

**Pull request merged**